keps/sig-apps/2804-consolidate-workload-controllers-status/README.md

```diff
@@ -113,9 +122,14 @@ Workload controllers should have above conditions (when applicable) to reflect t
 As an end-user of Kubernetes, I'd like all my workload controllers to have consistent statuses.
```

Comment:
This is not really a story per se - I'd remove this.  I'd start with a story about 

> As a user deploying workloads to Kubernetes, I would like to know how to correctly wait (via observation, CLI, or client code) for a newly created or recently update deployment, statefulset, or daemonset to be rolled out and at the correct version.

(which will require a combo of conditions, describing observed generation / generation, and nuances of each controller)

I would also change story 2 to be:

> As a developer building workload orchestration (operators, gitops, higher level workload types) above Kubernetes, I would like to know generically when an update to a service workload is fully rolled out and safe to declare as successful, and know what guidance to give workload creators around common failures

(that implies the Available condition or a replacement as above, as well as explaining Progressing and the progress deadlines)

For 3 and 4 I think of those as variants of the first two stories, but the goal isn't "unified statuses", the goal is "each class of workload controller has a reasonably consistent process".  Many of the questions asked here should be presented in the KEP as questions an end user might have:

* What happens if I try to wait for Available on a paused controller?
* Should pausing cause progressing to be false?
* What intermittent errors should be surfaced in the conditions and how do we avoid clients having to add more and more tests for intermittent errors (answer: by having the condition abstract them)?

We should also formally describe the class of workloads in the KEP (as definitions) - service and batch are two starts, but we need to do a quick pass to determine what other workload classes might have need of this (by reviewing examples from the extension ecosystem).

Reply:
I want to review the non-goals once the previous comment is addressed, but to propose a new condition we have to set overall objectives.  The first and foremost is that we understand the full process (how a user waits for a desired outcome, and how we define that outcome).  Once we have that, we then need to assess how to get there, and the KEP needs to reason from current state and then discuss the choices.  If Available is broken, then we have to figure out how it is currently useful to that flow (and what users may be expecting it to work and not getting the outcome they want), and capture that utility in the new outcome.

The first two non-goals I think are wrong - backward API compatibility is assumed and so we don't need to state it (or if we do, it needs to be very explicitly described in alternatives as why we chose not to break it).

Reply:
thanks for the review, will try to address it next week